packages/core/admin/admin/src/content-manager/pages/App/useContentManagerInitData.js

```diff
@@ -67,7 +59,13 @@ const useModels = () => {
         models,
       });
 
-      const actionToDispatch = setContentTypeLinks(ctLinks, stLinks, models, components);
+      const actionToDispatch = setInitData({
+        authorizedCtLinks: ctLinks,
+        authorizedStLinks: stLinks,
```

Comment:
This is sending me into another rabbit hole 😅 very much needed though 👍

Reply:
You can check [this commit](https://github.com/strapi/strapi/pull/16387/commits/77e05c260a137b23a6e5b80b64d05b0783543760) for the result. I tried to follow the thread of where `ct` and `st` were used, but I had to stop at [the hooks waterfalls](https://github.com/strapi/strapi/blob/77e05c260a137b23a6e5b80b64d05b0783543760/packages/core/admin/admin/src/content-manager/pages/App/useContentManagerInitData.js#L53-L61), because they could be used by other plugins, so I didn't want to cause breaking changes.

I also found on the way that the tests for getContentTypeLinks weren't running and were obsolete, so I did my best to fix that.

Finally, expanding the `ct` and `st` meant I had to choose between the singular and plural for their names. The app references them using both singular and plural in different places, which made this confusing. I switched as many things to singular as possible for consistency, but I can do the same towards the plural if needed.